File: frontend/src/types/tradeMatch.ts
export interface DuplicateCard {
  card_id: number;
  name_en: string | null;
  name_pt: string | null;
  set_code: string | null;
  collector_number: string | null;
  quantity: number;
  surplus: number;
  quality: string | null;
  image_uri: string | null;
  current_price: number | null;
}

export interface MatchedCard {
  card_id: number;
  name_en: string;
  set_code: string | null;
  image_uri: string | null;
  partner_quantity: number;
  your_max_price: number | null;
}

export interface TradeMatch {
  partner_name: string;
  share_code: string;
  matching_card_count: number;
  matched_cards: MatchedCard[];
}

File: src/api/schemas/trade_match.py
"""Trade matching API schemas (F110-T03/T04)."""

from __future__ import annotations

from pydantic import BaseModel


class DuplicateCard(BaseModel):
    card_id: int
    name_en: str | None
    name_pt: str | None = None
    set_code: str | None = None
    collector_number: str | None = None
    quantity: int
    surplus: int
    quality: str | None = None
    image_uri: str | None = None
    current_price: float | None = None


class MatchedCard(BaseModel):
    card_id: int
    name_en: str
    set_code: str | None = None
    image_uri: str | None = None
    partner_quantity: int
    your_max_price: float | None = None


class TradeMatch(BaseModel):
    partner_name: str
    share_code: str
    matching_card_count: int
    matched_cards: list[MatchedCard]
